quint/src/types/aliasInliner.ts

```diff
@@ -0,0 +1,87 @@
+/* ----------------------------------------------------------------------------------
+ * Copyright (c) Informal Systems 2023. All rights reserved.
+ * Licensed under the Apache 2.0.
+ * See License.txt in the project root for license information.
+ * --------------------------------------------------------------------------------- */
+
+/**
+ * Inlining for type aliases, to be used when resolving type aliases with the
+ * lookup table is not feasible.
+ *
+ * @author Gabriela Moreira
+ *
+ * @module
+ */
+
+import { IRTransformer, transformDefinition, transformModule, transformType } from '../IRTransformer'
+import { LookupTable } from '../names/lookupTable'
+import { QuintDef, QuintModule } from '../quintIr'
+import { QuintType } from '../quintTypes'
+
+/**
+ * Replaces all type aliases in a QuintModule with their resolved types, using
+ * the provided LookupTable. Uninterpreted types are left unchanged.
+ *
+ * @param lookupTable - The LookupTable containing the type aliases to be
+ * resolved.
+ * @param quintModule - The QuintModule to transform.
+ *
+ * @returns The transformed QuintModule with all type aliases replaced with
+ * their resolved types.
+ */
+export function inlineAliases(lookupTable: LookupTable, quintModule: QuintModule): QuintModule {
+  const inliner = new AliasInliner(lookupTable)
+  return transformModule(inliner, quintModule)
+}
+
+/**
+ * Inlines type aliases in a QuintType using the provided LookupTable.
+ *
+ * @param lookupTable - The LookupTable containing the type aliases to be
+ * resolved.
+ * @param type - The QuintType to transform.
+ *
+ * @returns The transformed QuintType with all type aliases replaced with
+ * their resolved types.
+ */
+export function inlineAliasesInType(lookupTable: LookupTable, type: QuintType): QuintType {
```

Comment:
sim

```suggestion
export function inlineAliasesInType(type: QuintType, lookupTable: LookupTable): QuintType {
```

Reply:
I think it makes sense. I was sort of trying to use this pattern that comes from languages with currying, where the most "fixed" arguments appear first and the most varying, later. But to be honest, it's kind of a mess right now where some parts use that, and some parts use most-important-first.

I guess most-important-first makes more sense in a language without currying, so let's go with that :)